tests/trainer/test_trainer.py

```diff
@@ -137,6 +141,8 @@ def config(self, device, precision, world_size):
     @pytest.fixture(autouse=True)
     def create_reference_model(self, config, tmpdir_factory, *args):
         """Trains the reference model, and saves checkpoints."""
+        config = deepcopy(config)
```

Comment:
If the config is test-specific (function scoped), can you elaborate why it needs to be deep-copied? I must be missing something here...

Reply:
For each test, `config` gets run, and then passed to both this fixture and the test itself. If `config` is not deepcopied, then the reference model and the test point to the same model object, and therefore the test always passes.

Reply:
Ah, I see. I missed this was in a fixture. Maybe could you add this in a comment?